File: c/src/Behavioral/Interpreter/TitleAndActor.c


//TitleAndActor - A Helper Class

#include "TitleAndActor.h"
#include "mem.h"
#include "assert.h"

TitleAndActor_t * TitleAndActor_new( char * title, char * actor)
{
	assert(title);
	assert(actor);
	TitleAndActor_t * ta;
	NEW(ta);

	ta->title = title;
	ta->actor = actor;
	return ta;
}

void TitleAndActor_free(  TitleAndActor_t * ta )
{
	FREE(ta);
}

File: javascript/src/Behavioral/Interpreter/TitleAndActor.js
(function() {
  define([], function() {
    var TitleAndActor;
    return TitleAndActor = (function() {
      function TitleAndActor(title, actor) {
        this.title = title;
        this.actor = actor;
      }

      TitleAndActor.prototype.getTitle = function() {
        return this.title;
      };

      TitleAndActor.prototype.getActor = function() {
        return this.actor;
      };

      return TitleAndActor;

    })();
  });

}).call(this);
